### PR TITLE
SERVER-84403 Add complex $in queries

### PR DESCRIPTION
**Pull request by henrinikku**
Thanks for submitting a PR to the Genny repo. Please include the following fields (if relevant) prior to submitting your PR.

**Jira Ticket:** SERVER-84403

**Whats Changed:**  
Extend existing collscan benchmarks with $in queries.

**Patch testing results:**  
[Patch](https://spruce.mongodb.com/version/65a81f7c3627e0d7d5f5d4b4/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC)

### CODE REVIEW

File: src/phases/query/CollScanComplexPredicate.yml
Comment (alyacb) [inline src/phases/query/CollScanComplexPredicate.yml:33]: I'd be surprised if this successfully executes on the large version of this workload, given that 100k docs was the limit for the microbenchmark workloads. Same concern for the other workload as well.

File: src/phases/query/CollScanComplexPredicate.yml
Comment (alyacb) [inline src/phases/query/CollScanComplexPredicate.yml:82]: Is the only difference here from the MixedQuery3LevelsDeep test the addition of the VeryLargeIn clause? If so, maybe add a comment about it? Same goes for below.

File: src/phases/query/CollScanComplexPredicate.yml
Comment (alyacb) [inline src/phases/query/CollScanComplexPredicate.yml:598]: We're not getting the same In every time are we?

File: src/phases/query/CollScanComplexPredicate.yml
Comment (alyacb) [inline src/phases/query/CollScanComplexPredicate.yml:682]: Were the fields with Ins here chosen so as not to be simplifiable?

**Review by alyacb — CHANGES_REQUESTED** [review]
Great work! I do have a few questions before I LGTM.

File: src/phases/query/CollScanComplexPredicate.yml
Comment (henrinikku) [reply src/phases/query/CollScanComplexPredicate.yml:33]: I was expecting it to fail at least on some variant as well but it didn't ([patch](https://spruce.mongodb.com/version/65a81f7c3627e0d7d5f5d4b4/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&taskName=coll_scan)). I'll run a couple more patches tomorrow to confirm it doesn't randomly fail.

**Review by henrinikku — COMMENTED** [review]
Thanks Alya! I addressed all comments.

**Review by alyacb — APPROVED** [review]
LGTM, pending patch results!

**Comment by henrinikku** [comment]
Looks like `ElemMatchVeryLargeIn` failed once in 8 runs on the sampling ce variant ([parsley link](https://parsley.mongodb.com/evergreen/sys_perf_linux_standalone_sampling_bonsai.2022_11_coll_scan_complex_predicate_large_patch_8a235a4547736605a98c44a6978010e989a5d826_65aa509a32f417420bdbb6fc_24_01_19_10_39_02/0/task?bookmarks=0,14523&shareLine=1425)). I decreased the $in list size for that query and restarted the patches.

**Comment by henrinikku** [comment]
That fixed the previously failing query,  but now `ElemMatchDNFWithLargeIns` is failing some of the time (it seems the previous failures were hiding this one). I simplified the query a bit and started another [8 patches](https://spruce.mongodb.com/user/henri.nikku/patches?page=0&patchName=22-jan%20in%20benchmark).

**Comment by henrinikku** [comment]
All patches succeeded. I'll go ahead and merge.

**Pull request merged**
